overlord/configstate/config/transaction.go

```diff
@@ -143,6 +143,18 @@ func (t *Transaction) GetMaybe(snapName, key string, result interface{}) error {
 }
 
 func getFromPristine(snapName string, subkeys []string, pos int, config map[string]*json.RawMessage, result interface{}) error {
+	// special case - get root document
+	if len(subkeys) == 0 {
+		if len(config) == 0 {
+			return fmt.Errorf("snap %q has no configuration", snapName)
```

Comment:
I think the special casing needs to be in NoOptionError, not here. We do want callers to be able to tell that an error is about the fact configuration wasn't available instead of something else more serious.

Reply:
Ok, I'm not 100% sure I got your suggestion right, but I special-cased the error message in NoOptionError Error() handler and throw NoOptionError from here.